packages/enterprise/src/license.ts

```diff
@@ -275,3 +267,33 @@ export async function setLicense(l: LicenseData | null) {
   // make sure we trust that l is already verified before setting:
   licenseData = l;
 }
+
+/**
+ * Checks if the license is expired.
+ * @returns {boolean} True if the license is expired, false otherwise.
+ */
+export function licenseIsExpired(): boolean {
+  // If licenseData is not available, consider it as not expired
+  if (!licenseData) {
+    return false;
+  }
+
+  // If expiration date is provided in the license data
+  if (licenseData.exp) {
+    // Create a date object for the expiration date
+    const expirationDate = new Date(licenseData.exp);
+
+    // Add a buffer of 2 days to the expiration date
+    const expiresWithBuffer = new Date(expirationDate);
+    expiresWithBuffer.setDate(expirationDate.getDate() + 2);
+
+    // Check if the adjusted expiration date is in the past
+    if (expiresWithBuffer < new Date()) {
+      // The license is expired
+      return true;
+    }
+  }
```

Comment:
From my understanding before if it was a `.trial` license and it expired we errored immediately with no buffer.  It errored too harshly to the point that the app became unusable.  This change would give them the two day buffer as well, but I think we only want to have the two day grace period with enterprise customers.

Reply:
Yeah, as it is, it adds the buffer for all users. I'll make sure it only does so for non-trial users.